be/tool/beck.c

```diff
@@ -748,14 +852,23 @@ static int recdo(struct scanner *s, const struct m0_format_tag *tag,
 	result = get(s, buf, size);
 	if (result == 0) {
 		if (memcmp(tag, &r->r_tag, sizeof *tag) == 0) {
+			/**
+			 * Check generation identifier before format footer
+			 * verification. Only process the records whose
+			 * generation identifier matches or is within
+			 * +/- MAX_GEN_DIFF_SEC seconds of segment's generation
+			 * identifier.
+			 */
+			if (r->r_ops != NULL && r->r_ops->ro_check != NULL) {
```

Comment:
Add this as M0_PRE() in this function.

Reply:
This cannot be M0_PRE because (r->r_ops == NULL) is valid for some record types.